packages/medusa/src/services/event-bus.ts

```diff
@@ -192,9 +201,15 @@ export default class EventBusService {
       })
       return await stagedJobRepository.save(stagedJobInstance)
     } else {
-      const opts: { removeOnComplete: boolean; delay?: number } = {
+      const opts: { removeOnComplete: boolean } & EmitOptions = {
```

Comment:
todo/suggestion(if-minor): Move `removeOnComplete` to `EmitOptions` and default value to `options` parameter instead of here